--- /usr/local/lib/python3.7/site-packages/airflow/bin/cli.py	2020-03-10 15:44:04.966478109 +0100
+++ /usr/local/lib/python3.7/site-packages/airflow/bin/cli.py	2020-03-19 16:29:00.591169931 +0100
@@ -1176,7 +1176,7 @@
     elif url.get_backend_name() == 'postgresql':
         env = os.environ.copy()
         env['PGHOST'] = url.host or ""
-        env['PGPORT'] = url.port or ""
+        env['PGPORT'] = str(url.port or "")  # JJM patched, env vars should be strings
         env['PGUSER'] = url.username or ""
         # PostgreSQL does not allow the use of PGPASSFILE if the current user is root.
         env["PGPASSWORD"] = url.password or ""
