--- /usr/local/lib/python3.9/site-packages/airflow/www/views.py 2022-01-19 10:47:20.035044670 +0000
+++ /usr/local/lib/python3.9/site-packages/airflow/www/views.py 2022-01-19 10:50:02.584152438 +0000
@@ -849,11 +849,18 @@
         # Filter to only ask for accessible and selected dags
         query = query.filter(DagRun.dag_id.in_(filter_dag_ids))  # pylint: enable=no-member

+        # Patched version from upstream Airflow
+        # https://github.com/apache/airflow/pull/18315/commits/f9f499f1aeaed774f08a55f738c10856dd4c696c
+        def _datetime_to_string(value: Optional[DateTime]) -> Optional[str]:
+            if value is None:
+                return None
+            return value.isoformat()
+
         resp = {
             r.dag_id.replace('.', '__dot__'): {
                 'dag_id': r.dag_id,
-                'execution_date': r.execution_date.isoformat(),
-                'start_date': r.start_date.isoformat(),
+                'execution_date': _datetime_to_string(r.execution_date),
+                'start_date': _datetime_to_string(r.start_date),
             }
             for r in query
